hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/DefaultSource.scala

```diff
@@ -105,7 +105,9 @@ class DefaultSource extends RelationProvider
     val tableType = metaClient.getTableType
     val queryType = parameters(QUERY_TYPE_OPT_KEY)
     log.info(s"Is bootstrapped table => $isBootstrappedTable, tableType is: $tableType")
-
+    val schemaUtil = new TableSchemaResolver(metaClient)
```

Comment:
possible to add some tests?

Reply:
I did think about it, but not easy. basically we need to fail the first write and then trigger a read. but all these need to happen at data source layer. and hence can't use test table.

Reply:
why though? we can create the using the test table API and read from data source right?